### ISSUE TITLE
Add grid search and plot to SimpleNN and GroupsNN

### ISSUE THREAD

**Issue by AgathaSchmidt** [SciCompMod/memilio#1137]
### Motivation / Current Behaviour

Right now, we only provide the data gernation as well as the final neural network based model for the surrogate models for one population in the main branch. 

### Enhancement description

Now we decided that we want to provide the code for finding the final architecture via a grid search etc. and the code for the visualizations. 

- [ ] grid search (layers, units per layer)
- [ ] hyperparameter testing 
- [ ] plots 

### Checklist

- [X] Attached labels, especially loc:: or model:: labels.
- [X] Linked to project

**Comment by mknaranja** [comment]
Outdated
